tokio-timer/src/timer/mod.rs

```diff
@@ -124,7 +123,7 @@ use tokio_executor::park::{Park, ParkThread, Unpark};
 /// [`turn`]: #method.turn
 /// [Handle.struct]: struct.Handle.html
 #[derive(Debug)]
-pub struct Timer<T, N = SystemNow> {
+pub struct Timer<T, N = Clock> {
```

Comment:
Sorry I wasn't clear enough, I meant we can remove the generic completely from `Timer`, and just use `clock::now()` inside.

Reply:
Thank you for the clarification, will do! :)

Reply:
Sorry to bother again, but I wan't to be sure and do it right.

Will the `Timer::new_with_now(...)` function stay? (I am not sure because of the wording in the issue) It is used in tests, and besides `pub` I think you wan't to keep it
Also do you mean the Clock-crate `clock::now()` function or the Clock crate struct `Clock::now()` function?

I feel a little overwhelmed, I don't wan't make an easy issue to an issue you have to guide me too much!
But I'll push what I think you want anyways.

Reply:
No worries!

`Timer::new_with_now` should be deleted. Tests that use `new_with_now` can be updated to use the [mocking API from `tokio-test` ](https://github.com/tokio-rs/tokio/blob/master/tokio-test/src/clock.rs).

It does look like there is some usage of `new_with_now` else where. In *theory* it should just be changed to `new`, **however** I see one location where `clock::with_default` is missing.

[here](https://github.com/tokio-rs/tokio/blob/master/tokio/src/runtime/threadpool/background.rs#L33-L34) wrap those two lines w/ `clock::with_default(clock, ...)`.

That _should_ be good.